beacon-chain/blockchain/service.go

```diff
@@ -299,7 +325,63 @@ func (c *ChainService) processBlock(block *types.Block) error {
 	return nil
 }
 
-func (c *ChainService) executeStateTransition(
+func (c *ChainService) processBlock(block *types.Block) error {
+
+	blockhash, err := block.Hash()
+	if err != nil {
+		return fmt.Errorf("could not hash incoming block: %v", err)
+	}
+
+	beaconState, err := c.beaconDB.GetState()
+	if err != nil {
+		return fmt.Errorf("failed to get beacon state: %v", err)
+	}
+
+	if block.SlotNumber() == 0 {
+		return errors.New("cannot process a genesis block: received block with slot 0")
+	}
+
+	// Save blocks with higher slot numbers in cache.
+	if !c.isBlockReadyForProcessing(block) && block.SlotNumber() > beaconState.Slot() {
+		c.unProcessedBlocks[block.SlotNumber()] = block
+		return fmt.Errorf("block with hash %#x is not ready for processing", blockhash)
```

Comment:
Is this really an error? What about when we are not yet in sync, but peers send future blocks to us while we are downloading new blocks? I'm assuming this error message is going to be visible in the logs, but unnecessarily.

I think this is expected behavior, not necessarily an error. What do you all think?

Reply:
Yeap you are right, this shouldnt be an error, I will change this to a debug log